### PR TITLE
create game object and piece objects

### PR DESCRIPTION
**Pull request by erinmadisonanthony**
I took a stab at this but I strongly suspect that changes need to be made prior to merging. Please let me know how I should alter this. Thanks.

### CODE REVIEW

File: app/models/piece.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 class Piece < ActiveRecord::Base
-
+  belongs_to :game
```
Comment (hoitomt) [inline app/models/piece.rb]: This is correct. In order to make this work we need to add a game_id column to the pieces table.

```
rails generate migration add_game_id_to_pieces
```

Update your migration file to look like this:
```
class AddGameIdToPieces < ActiveRecord::Migration
  def change
    add_column :pieces, :game_id, :integer
  end
end
```

File: app/models/game.rb [hunk 2]
```diff
@@ -3,4 +3,52 @@ class Game < ActiveRecord::Base
...
+  has_many :pieces
+
+  def self.create_game(game_params)
+    g = Game.create(game_params)
+    g.create_pieces
+    return g 
+  end
+
+  def create_pieces
+    Piece.create(:type => 'Pawn', :x_position => 0, :y_position => 1)
```
Comment (hoitomt) [inline app/models/game.rb]: Each piece should have a game and a user on it. Each one should look like this (or user black_player):

```
Piece.create(:type => 'Pawn', :x_position => 0, :y_position => 1, user: white_player, game: self)
```

**Comment by hoitomt** [comment]
@erinmadisonanthony this is pretty close - just a couple of changes needed

**Comment by hoitomt** [comment]
:+1: Looks good! Please make sure the create_game method works before merging.

```
rails console
Piece.count #this will return a count
params = {name: 'Make some pieces', white_player: User.first, black_player: User.last}
g = Game.create_game(params)
Piece.count # This should be the original Piece.count + 32
```

**Comment by wookee55** [comment]
it seems like the create_game method isn't be called anywhere? should we be calling the create_game method in the create action in the games controller instead of create?

**Comment by hoitomt** [comment]
@wookee55 yes, we will make that change eventually

**Pull request merged**